src/Sarif.Viewer.VisualStudio.ResultSources.Factory.Core/ResultSourceFactory.cs

```diff
@@ -29,9 +29,9 @@ public class ResultSourceFactory : IResultSourceFactory
     {
         private readonly StandardKernel standardKernel;
         private readonly string solutionRootPath;
-        private readonly List<Type> resultSourceTypes = new List<Type>
+        private readonly Dictionary<Type, (int, int)> resultSources = new Dictionary<Type, (int firstMenuId, int firstCommandId)>
```

Comment:
>(int, int) [](http://example.com/codeflow?start=42&length=10)

maybe create a new class/struct for the combination of menu id + command id? its also used in menu event args, and will be more readable

Reply:
I'd prefer to keep it this way, even though I agree it's a bit more work to parse visually. On the other hand, you get to see the unpacked property names, which helps with comprehension.
This approach also avoids having to deal with IEqualityComparer.

Reply:
make sense